src/heretic/model.py

```diff
@@ -111,48 +155,47 @@ def reload_model(self):
             self.trusted_models[self.settings.model] = True
 
     def get_layers(self) -> ModuleList:
+        model = self.model
+        
+        # Unwrap PeftModel
+        if isinstance(model, PeftModel):
+            model = model.base_model.model
+
         # Most multimodal models.
         with suppress(Exception):
-            return self.model.model.language_model.layers
+            return model.model.language_model.layers
 
         # Text-only models.
-        return self.model.model.layers
+        return model.model.layers
 
-    def get_layer_matrices(self, layer_index: int) -> dict[str, list[Tensor]]:
+    def get_layer_modules(self, layer_index: int) -> dict[str, list[torch.nn.Module]]:
```

Comment:
```suggestion
    def get_layer_modules(self, layer_index: int) -> dict[str, list[torch.nn.Module | Tensor]]:
```

I intend to enforce type checks soon so we should be precise.